core/shared/src/main/scala-2.11-2.12/ChunkLike.scala

```diff
@@ -0,0 +1,187 @@
+    }
+
+  /**
+   * Returns the first index for which the given predicate is satisfied.
+   */
+  override final def indexWhere(f: A => Boolean): Int =
+    indexWhere(f, 0)
+
+  /**
+   * Returns a chunk with the elements mapped by the specified function.
+   */
+  override final def map[B, That](f: A => B)(implicit bf: CanBuildFrom[Chunk[A], B, That]): That =
+    bf match {
+      case _: ChunkCanBuildFrom[_] => mapChunk(f)
+      case _                       => super.map(f)
+    }
+
+  /**
+   * Generates a readable string representation of this chunk using the
+   * specified start, separator, and end strings.
+   */
+  override final def mkString(start: String, sep: String, end: String): String = {
+    val builder = new scala.collection.mutable.StringBuilder()
+
+    builder.append(start)
+
+    var i   = 0
+    val len = self.length
+
+    while (i < len) {
+      if (i != 0) builder.append(sep)
+      builder.append(self(i).toString)
+      i += 1
+    }
+
+    builder.append(end)
+
+    builder.toString
+  }
+
+  /**
+   * Generates a readable string representation of this chunk using the
+   * specified separator string.
+   */
+  override final def mkString(sep: String): String =
+    mkString("", sep, "")
+
+  /**
+   * Generates a readable string representation of this chunk.
+   */
+  override final def mkString: String =
+    mkString("")
+
+  /**
+   * Determines if the chunk is not empty.
+   */
+  override final def nonEmpty: Boolean =
+    length > 0
+
+  /**
+   * The number of elements in the chunk.
+   */
+  override final def size: Int =
+    length
+
+  /**
+   * The implementation of `flatMap` for `Chunk`.
+   */
+  protected final def flatMapChunk[B, That](f: A => GenTraversableOnce[B]): Chunk[B] = {
+    val len                    = self.length
+    var chunks: List[Chunk[B]] = Nil
+
+    var i               = 0
+    var total           = 0
+    var B0: ClassTag[B] = null.asInstanceOf[ClassTag[B]]
+    while (i < len) {
+      val chunk = f(self(i)) match {
+        case chunk: Chunk[B] => chunk
+        case other           => Chunk.fromIterable(other.toList)
+      }
+
+      if (chunk.length > 0) {
+        if (B0 == null)
+          B0 = Chunk.classTagOf(chunk)
+
+        chunks ::= chunk
+        total += chunk.length
+      }
+
+      i += 1
+    }
+
+    if (B0 == null) Chunk.empty
+    else {
+      implicit val B: ClassTag[B] = B0
+
+      val dest: Array[B] = Array.ofDim(total)
+
+      val it = chunks.iterator
+      var n  = total
+      while (it.hasNext) {
+        val chunk = it.next
+        n -= chunk.length
+        chunk.toArray(n, dest)
+      }
+
+      Chunk.arr(dest)
+    }
+  }
+
+  /**
+   * Constructs a new `ChunkBuilder`. This operation allocates mutable state
+   * and is not referentially transparent. It is provided for compatibility
+   * with Scala's collection library and should not be used for other purposes.
+   */
+  override protected[this] def newBuilder: ChunkBuilder[A] =
```

Comment:
By convention, side-effecting methods should have nullary parameter list: `def newBuilder()`.

Reply:
Yes. The signature of the `newBuilder` method we are overriding doesn't have the nullify parameter list. Let me see if I can add and still have it be recognized as a valid override.

Reply:
Yes unfortunately adding the nullary parameter list generate a compiler error so I think we will have to leave as is.